drivers/flash/flash_rpi_pico.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <hardware/flash.h>
+
+LOG_MODULE_REGISTER(flash_rpi, CONFIG_FLASH_LOG_LEVEL);
+
+#define DT_DRV_COMPAT raspberrypi_pico_flash_controller
+
+#define PAGE_SIZE 256
+#define ERASE_VALUE 0xff
+#define FLASH_PAGE_MASK (0xffffffff - (PAGE_SIZE - 1))
+#define FLASH_SIZE (CONFIG_FLASH_SIZE * 1024)
+
+static uint8_t page_buffer[PAGE_SIZE];
+
+static const struct flash_parameters flash_rpi_parameters = {
+	.write_block_size = 1,
+	.erase_value = ERASE_VALUE,
+};
+
+static bool is_invalid_range(off_t offset, uint32_t size)
+{
+	return offset + size > FLASH_SIZE;
+}
+
+static int flash_rpi_read(const struct device *dev, off_t offset,
+				void *data, size_t size)
+{
+	if (is_invalid_range(offset, size)) {
+		return -EINVAL;
+	}
+
+	if (!size) {
```

Comment:
MISRA C:2004, 13.2 - Tests of a value against zero should be made explicit, unless the operand is effectively Boolean.